mgr/manager.go

```diff
@@ -314,34 +311,22 @@ func New(dir, remote, svcmgr, before, interval string) (*Manager, error) {
 		Interval:       interval,
 	}
 
-	return setup(m)
+	return m, m.validate()
 }
 
 // setup provides the common final setup work that needs to be done
 // for a Manager to be ready.
-func setup(m *Manager) (*Manager, error) {
-	var err error
-
+func (m *Manager) validate() error {
+	if m.Dir == "" {
+		return fmt.Errorf("manager: invalid manager configuration (missing spec dir)")
+	}
 	m.Dir = filepath.Clean(m.Dir)
 
 	if m.ServiceManager == "" {
 		m.ServiceManager = "dummy"
```

Comment:
This isn't new code so don't worry about it, but it's a code smell to me to see a validation function modifying things.

Reply:
In this case it's setting a default, but generally agreed- the pathways involved here still need some cleanup.